massHistsOptimal.C

```diff
@@ -0,0 +1,165 @@
+   double yields_signal[100];
+   double yields_bkg[100];
+   double mvaCuts[100];
+   for(int iMva=0; iMva<13; iMva++){
+   //for(int iMva=2; iMva<4; iMva++){
+
+      //create Pic/mva to store pictures
+      std::string dirMvaPic(TString::Format( "if [ ! -d \"%sPic/pT%.1f-%.1f_y%.1f-%.1f/mva%d\" ]; then\n"
+                                 "    mkdir %sPic/pT%.1f-%.1f_y%.1f-%.1f/mva%d \n"
+                                 "fi", ana::whichtree[mode].c_str(), ana::pTMin,ana::pTMax,ana::yMin,ana::yMax, iMva, 
+                                 ana::whichtree[mode].c_str(), ana::pTMin,ana::pTMax,ana::yMin,ana::yMax, iMva));
+      gSystem->Exec(dirMvaPic.c_str());
+
+      double mvaCut = (double)iMva*ana::mvaStep + ana::mvaMin;
+
+      // extract invariant mass
+      int mvaBinMin = hDcaVsMassAndMvaNPD0["h_match_all"]->GetYaxis()->FindBin(mvaCut+0.1*ana::mvaStep); // 0.1 offset to make sure the correct bin is returned
+      int mvaBinMax = hDcaVsMassAndMvaNPD0["h_match_all"]->GetYaxis()->GetNbins()+1;
+      int label = mvaCut*100;
+      TH1D* hMassNPD0 = hDcaVsMassAndMvaNPD0["h_match_unswap"]->ProjectionX(Form("hMassMCNPD0mva%d", label), mvaBinMin, mvaBinMax);
+      TH1D* hMassNPD0All = hDcaVsMassAndMvaNPD0["h_match_all"]->ProjectionX(Form("hMassMCNPD0Allmva%d", label), mvaBinMin, mvaBinMax);
+      TH1D* hMassPD0 = hDcaVsMassAndMvaPD0["h_match_unswap"]->ProjectionX(Form("hMassMCPD0mva%d", label), mvaBinMin, mvaBinMax);
+      TH1D* hMassPD0All = hDcaVsMassAndMvaPD0["h_match_all"]->ProjectionX(Form("hMassMCPD0Allmva%d", label), mvaBinMin, mvaBinMax);
+      TH1D* hMassData = hDcaVsMassAndMvaDataD0->ProjectionX(Form("hMassDataD0mva%d", label), mvaBinMin, mvaBinMax);
+
+      hMassNPD0->Scale(1.0/hMassNPD0->GetBinWidth(1));
+      hMassNPD0All->Scale(1.0/hMassNPD0All->GetBinWidth(1));
+      hMassPD0->Scale(1.0/hMassPD0->GetBinWidth(1));
+      hMassPD0All->Scale(1.0/hMassPD0All->GetBinWidth(1));
+      hMassData->Scale(1.0/hMassData->GetBinWidth(1));
+
+      // TFitResultPtr constructed by std::shared_ptr<TFitResult>, be free of it even if you do not delete it
+      // fitting the mass
+      // rescale the mass
+      
+      TFitResultPtr fitResultPtr;
+      TF1 fMass;
+      if(mode == 0) fMass = massfitting(hMassData, hMassPD0, hMassPD0All, TString::Format("fMassMva%d", iMva), fitResultPtr);
+      if(mode == 1) fMass = massfitting(hMassData, hMassNPD0, hMassNPD0All, TString::Format("fMassMva%d", iMva), fitResultPtr);
+      if(mode == 2) fMass = massfitting(hMassData, hMassNPD0, hMassNPD0All, TString::Format("fMassMva%d", iMva), fitResultPtr);
+
+      // draw the fitting
+      drawMassFitting(hMassData, fMass, TString::Format("%sPic/pT%.1f-%.1f_y%.1f-%.1f/mva%d/MassFittingUnNormalized.png", ana::whichtree[mode].c_str(), ana::pTMin,ana::pTMax,ana::yMin,ana::yMax, iMva), 
+            TString::Format("%.1f<p_{T}<%.1fGeV, %.1f<y<%.1f",ana::pTMin,ana::pTMax,ana::yMin,ana::yMax), TString::Format("MVA > %.2f", 0.4+0.02*iMva));
```

Comment:
line 51 defines mvaCut = iMva * mvaStep + mvaMin, min is not 0.4 anymore in the current version of myAnaConsts.h